sdk/rust/oak_runtime_compat/src/asylo/mutex.rs

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright 2019 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// See rust/src/libstd/sys/unix/mutex.rs
```

Comment:
Consider pointing to a permalink here, e.g. https://github.com/rust-lang/rust/blob/696735f71b4408302ba166d148e9d474c51416d2/src/libstd/sys/unix/mutex.rs , and describe at high level what modifications were necessary, otherwise it will be quite hard to keep this in sync. Even better, add some comments near the parts of this file that differ from upstream, if it's practical.

Reply:
Ive added perma links for mutex, error, io/error. `src/asylo/thread.rs` has some high level notes. Its going to painful to keep this in sync with upstream changes, but the original sources should be very stable.